### PR TITLE
[Program Capture] Capture & execute `qml.jacobian` in plxpr

### PR DESCRIPTION
**Pull request by dwierichs**
**Context:**
We're adding support for differentiation in plxpr, also see #6120.

**Description of the Change:**
This PR adds support for `qml.jacobian`, similar to the support for `qml.grad`.
Note that Pytree support will be needed to allow for multi-argument derivatives.

**Benefits:**
Capture derivatives of non-scalar functions.

**Possible Drawbacks:**
See discussion around `qml.grad` in #6120.

**Related GitHub Issues:**

[sc-71860]

### CODE REVIEW

File: pennylane/_grad.py [hunk 1]
```diff
@@ -434,7 +434,10 @@ def circuit(x):
         return ops_loader.jacobian(func, method=method, h=h, argnums=argnum)
 
     if method or h:
-        raise ValueError(f"Invalid values for 'method={method}' and 'h={h}' in interpreted mode")
+        raise ValueError(f"Invalid values '{method=}' and '{h=}' without QJIT.")
+
+    if enabled():
+        return _capture_diff(func, argnum, create_jacobian_primitive())
```
Comment (dwierichs) [inline pennylane/_grad.py:440]: Reusing the code for `qml.grad` by simply passing the different primitive.

**Comment by dwierichs** [comment]
@albi3ro Do we need tests for the higher-order primitives (like `adjoint`, `for_loop`...) with `qml.jacobian`, or is it fine to leave it at `grad`?

**Comment by albi3ro** [comment]
> @albi3ro Do we need tests for the higher-order primitives (like `adjoint`, `for_loop`...) with `qml.jacobian`, or is it fine to leave it at `grad`?

I think we should be ok leaving them out for now,  unless we have reason to believe they would have some sort of different behavior.

**Review by albi3ro — APPROVED** [review]
👍

**Review by mudit2812 — APPROVED** [review]
Looks great! Thanks for the very detailed comments in the tests, it made it a lot easier to follow along 😄

**Pull request merged**